### PR TITLE
Add CHANGELOG.md

### PR DESCRIPTION
**Pull request by songy23**

### CODE REVIEW

**Comment by songy23** [comment]
https://github.com/open-telemetry/opentelemetry-specification/pull/161/commits/de57e309c5f37733b2a80731a494f30cd7397041: copied from https://github.com/open-telemetry/opentelemetry-dotnet/blob/master/CHANGELOG.md.

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Changelog
+
+Please update changelog as part of any significant pull request. Place short
+description of your change into "Unreleased" section. As part of release
+process content of "Unreleased" section content will generate release notes for
+the release.
+
+## Unreleased
+
+## 0.1 (API proposal complete) - 06-21-2019
```
Comment (iredelmeier) [inline CHANGELOG.md]:
```suggestion
## 0.1.0 (API proposal complete) - 06-21-2019
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
```
Comment (iredelmeier) [inline CHANGELOG.md]:
```suggestion

### Added

```

**Comment by iredelmeier** [comment]
> @SergeyKanzhelev is this how we should structure the changelog?

I suggest we follow [Keep a Changelog](https://keepachangelog.com).

**Comment by reyang** [comment]
How about https://github.com/github-changelog-generator/github-changelog-generator?

**Comment by iredelmeier** [comment]
In my experience, generated changelogs tend to be a lot less human-friendly than manual changes unless there are strong enough commit messages and issue/PR titles... which is generally harder to achieve than the CHANGELOG line itself ;)

That said, I'd definitely be on board with something to automate a requirement for a CHANGELOG change in PRs, e.g., through [danger](https://danger.systems/).

**Pull request merged**